testinfra/modules/blockdevice.py

```diff
@@ -0,0 +1,220 @@
+        True
+
+        >>> host.block_device("/dev/sda").is_partition
+        False
+
+        """
+        return self.is_block_device and self._data['start_sector'] > 0
+
+    @property
+    def size(self):
+        """Return size if the device in bytes.
+
+           Return None if device is not a block device.
+
+        >>> host.block_device("/dev/sda1").size
+        512110190592
+
+        >>> print(host.block_device("/dev/console").size)
+        None
+
+        """
+        if not self.is_block_device:
+            return None
+        return self._data['size']
+
+    @property
+    def sector_size(self):
+        """Return sector size for the device in bytes or None.
+
+        >>> host.block_device("/dev/sda1").sector_szie
+        512
+
+        >>> print(host.block_device("/dev/console").sector_szie)
+        None
+
+        """
+        if not self.is_block_device:
+            return None
+        return self._data['sector_size']
+
+    @property
+    def block_size(self):
+        """Return block size for the device in bytes or None.
+
+        >>> host.block_device("/dev/sda").block_szie
+        4096
+
+        >>> print(host.block_device("/dev/console").block_size)
+        None
+
+        """
+        if not self.is_block_device:
+            return None
+        return self._data['block_size']
+
+    @property
+    def start_sector(self):
+        """Return start sector of the device on the underlaying device.
+
+           Usually the value is zero for full devices and is non-zero
+           for partitions.
+           Return None if device is not a block device.
+
+        >>> host.block_device("/dev/sda1").start_sector
+        2048
+
+        >>> host.block_device("/dev/md0").start_sector
+        0
+
+        >>> print(host.block_device("/dev/console").start_sector)
+        None
+
+        """
+        if not self.is_block_device:
+            return None
+        return self._data['sector_size']
+
+    @property
+    def is_writable(self):
+        """Return True if device is writable (have no RO status), or None.
+
+        >>> host.block_device("/dev/sda").is_writable
+        True
+
+        >>> host.block_device("/dev/loop1").is_writable
+        False
+
+        >>> host.block_device("/dev/console").is_writable
+        None
+
+        """
+        if not self.is_block_device:
+            return None
+        mode = self._data['rw_mode']
+        if mode == 'rw':
+            return True
+        if mode == 'ro':
+            return False
+        raise ValueError('Unexpected value for rw: %s' % mode)
+
+    @property
+    def ra(self):
+        """Return Read Ahead for the device in 512-bytes sectors or None.
+
+        >>> host.block_device("/dev/sda").ra
+        256
+
+        >>> print(host.block_device("/dev/console").ra)
+        None
+        """
+        if not self.is_block_device:
+            return None
+        return self._data['read_ahead']
+
+    @property
+    def _data(self):
```

Comment:
You can use the `@cached_property` decorator from `testinfra.utils` to avoid handling the `_data_cache` attribute.